### PR TITLE
Front-end/login-page

### PR DESCRIPTION
**Pull request by hbngo21**

### CODE REVIEW

File: libs/authentication/src/lib/components/login/login.component.ts [hunk 1]
```diff
@@ -1,15 +1,50 @@
...
+        .pipe(first())
+        .subscribe(
+          (response) => {
+            this.errorMessage = '';
+            this.cookieService.set('token', response.token, 1);
+            localStorage.setItem('userName', loginForm.value.userName);
+            localStorage.setItem('imageMaxSizes', response.imageMaxSizes);
+            localStorage.setItem('serverSettings', response.settings);
+            //this.adjustUploadSettings(response);
+            this.router.navigate(['/', 'camera']);
```
Comment (huyjack178) [inline libs/authentication/src/lib/components/login/login.component.ts:41]: Please help add this logic after login successfully
```
adjustUploadSettings(response) {
      localStorage.setItem('uploadSettings', JSON.stringify(this.uploadSettings));

      if (localStorage.getItem('uploadSettings')) {
        const uploadSettings = JSON.parse(localStorage.getItem('uploadSettings'));
        const serverSettings = JSON.parse(response.data.settings);
        uploadSettings.ftp.enabled = uploadSettings.ftp.enabled && serverSettings.ftp.enabled;
        uploadSettings.cloudinary.enabled = uploadSettings.cloudinary.enabled && serverSettings.cloudinary.enabled;
        uploadSettings.local.enabled = serverSettings.local.enabledHigh || serverSettings.local.enabledLow;
        localStorage.setItem('uploadSettings', JSON.stringify(uploadSettings));
      }
    },
```

File: libs/authentication/src/lib/components/login/login.component.ts [hunk 2]
```diff
@@ -1,15 +1,50 @@
...
+  constructor(
+    private readonly authService: AuthService,
+    private readonly router: Router,
+    private readonly cookieService: CookieService
+  ) {}
 
-  ngOnInit(): void {}
+  onSubmit(loginForm: NgForm) {
+    if (!loginForm.valid) {
+      this.errorMessage = 'Please enter valid username and password!';
```
Comment (huyjack178) [inline libs/authentication/src/lib/components/login/login.component.ts]: Please help me change to vietnamese

File: libs/authentication/src/lib/components/login/login.component.html [hunk 3]
```diff
@@ -1,17 +1,47 @@
...
+      fxLayoutAlign="space-around center"
+    >
+      <mat-form-field appearance="outline" [style.width.px]="450">
+        <mat-label>Username</mat-label>
+        <input matInput ngModel name="userName" required />
+      </mat-form-field>
+      <mat-form-field
+        appearance="outline"
+        [style.width.px]="450"
+        hintLabel="At least 4 characters"
```
Comment (huyjack178) [inline libs/authentication/src/lib/components/login/login.component.html]: Vietnamese as well

File: libs/authentication/src/lib/components/login/login.component.ts (hunk 2 above)
Comment (hbngo21) [reply libs/authentication/src/lib/components/login/login.component.ts]: Thank you for your comment. I fixed it.

File: libs/authentication/src/lib/components/login/login.component.html (hunk 3 above)
Comment (hbngo21) [reply libs/authentication/src/lib/components/login/login.component.html]: Thank you for your comment. I fixed it.

File: libs/authentication/src/lib/components/login/login.component.ts (hunk 1 above)
Comment (hbngo21) [reply libs/authentication/src/lib/components/login/login.component.ts:41]: Thank you for your comment.
I will handle this logic in another component.

**Pull request merged**